src/osd/PG.cc

```diff
@@ -4078,24 +4111,33 @@ void PG::replica_scrub(
     map, start, end, msg->deep, msg->seed,
     handle);
 
-  vector<OSDOp> scrub(1);
-  scrub[0].op.op = CEPH_OSD_OP_SCRUB_MAP;
-  hobject_t poid;
-  eversion_t v;
-  osd_reqid_t reqid;
-  MOSDSubOp *subop = new MOSDSubOp(
-    reqid,
-    pg_whoami,
-    spg_t(info.pgid.pgid, get_primary().shard),
-    poid,
-    0,
-    msg->map_epoch,
-    osd->get_tid(),
-    v);
-  ::encode(map, subop->get_data());
-  subop->ops = scrub;
-
-  osd->send_message_osd_cluster(subop, msg->get_connection());
+  if (HAVE_FEATURE(acting_features, SERVER_LUMINOUS)) {
+    MOSDRepScrubMap *reply = new MOSDRepScrubMap(
+      spg_t(info.pgid.pgid, get_primary().shard),
+      msg->map_epoch,
+      pg_whoami);
+    ::encode(map, reply->get_data());
```

Comment:
My main concern is that we're requiring specific OSD paths for this message to decode properly. If we stick the ScrubMap into a bufferlist owned by the message, then even though it's encoded/decoded by the OSD, we can change the order of things without breaking stuff (by, say, sticking new fields on the end of the Message that only one OSD is new enough to know about). Just give it a bufferlist that MOSDRepScrubMap encodes/decodes; that surely isn't noticeable in performance.